packages/office-ui-fabric-react/src/components/Pivot/Pivot.tsx

```diff
@@ -67,20 +67,22 @@ export class Pivot extends React.Component<IPivotProps, IPivotState> {
   public componentWillReceiveProps(nextProps: IPivotProps) {
     const links: IPivotItemProps[] = this._getPivotLinks(nextProps);
 
-    let selectedKey: string;
-    if (this._isKeyValid(nextProps.selectedKey)) {
-      selectedKey = nextProps.selectedKey;
-    } else if (this._isKeyValid(this.state.selectedKey)) {
-      selectedKey = this.state.selectedKey;
-    } else {
-      selectedKey = links[0].itemKey;
-    }
+    this.setState((prevState, props) => {
+      let selectedKey: string;
+      if (this._isKeyValid(nextProps.selectedKey)) {
+        selectedKey = nextProps.selectedKey;
+      } else if (this._isKeyValid(prevState.selectedKey)) {
+        selectedKey = prevState.selectedKey;
+      } else {
+        selectedKey = links[0].itemKey;
+      }
```

Comment:
while very unlikely, if links were empty, this would nullref. Maybe we should add

```ts
else if (links.length) {
...
}
```